templates/singleton/boil_queries.tpl

```diff
@@ -51,49 +51,68 @@ func mergeModels(tx boil.Executor, primaryID uint64, secondaryID uint64, foreign
 func deleteConflictsBeforeMerge(tx boil.Executor, conflict conflictingUniqueKey, primaryID uint64, secondaryID uint64) error {
 	conflictingColumns := strmangle.SetComplement(conflict.columns, []string{conflict.objectIdColumn})
 
-	if len(conflictingColumns) < 1 {
-		return nil
-	} else if len(conflictingColumns) > 1 {
-		return errors.Err("this doesnt work for unique keys with more than two columns (yet)")
+	selectParams := []interface{}{primaryID, secondaryID}
+	// 1 to 1 relationship check
+	isFKManyto1 := 1
+	if len(conflict.columns) == 1 && len(conflictingColumns) == 0 {
+		isFKManyto1 = 0
+		conflictingColumns = conflict.columns
+		selectParams = []interface{}{secondaryID}
```

Comment:
Just thought of a case not covered. If the secondary user has a row but the primary does not the secondary row will still be deleted. We need to also check that the primary has a row too in order for the secondary to be deleted, otherwise there is no conflict to resolve.